### PR TITLE
DW-3296 First deployment into DataWorks architecture

### PR DESCRIPTION
**Pull request by dan-hill2802**
Signed-off-by: Daniel Hill <danhill@digital.uc.dwp.gov.uk>

### CODE REVIEW

**Comment by dan-hill2802** [comment]
Pipeline deployed, but paused, to aid with review:
https://concourse.service.dw/teams/dataworks/pipelines/aws-concourse

File: Makefile [hunk 1]
```diff
@@ -32,5 +37,23 @@ bootstrap-terraform: ## Bootstrap local environment for first use
...
 		export AWS_REGION=$(aws_region); \
-		python bootstrap_terraform.py; \
+		python3 bootstrap_terraform.py; \
 	}
+
+.PHONY: bootstrap-terraform-dev
+bootstrap-terraform-dev: ## Bootstrap local environment for first use
+	make bootstrap-terraform
+	/usr/bin/sed -i '' 's|"default" ? "management-dev"|"default" ? "development"|g' terraform/deploy/terraform.tf
+	/usr/bin/sed -i '' 's|terraform/dataworks/aws-concourse.tfstate|terraform/dataworks/aws-concourse-${shell git rev-parse --abbrev-ref HEAD}.tfstate|g' terraform/deploy/terraform.tf
```
Comment (benj-fletch) [inline Makefile]: We should just use `sed` here? it might not be in `user/bin` on all users' PCs

File: terraform/modules/concourse_web/data.tf [hunk 2]
```diff
@@ -18,10 +18,10 @@ data "aws_ssm_parameter" "concourse_password" {
...
-data "aws_ssm_parameter" "concourse_cognito_client_secret" {
-  name = var.cognito.client_secret_ssm_parameter
-}
+//data "aws_ssm_parameter" "concourse_cognito_client_id" {
+//  name = var.cognito.client_id_ssm_parameter
+//}
+//
+//data "aws_ssm_parameter" "concourse_cognito_client_secret" {
+//  name = var.cognito.client_secret_ssm_parameter
+//}
```
Comment (benj-fletch) [inline terraform/modules/concourse_web/data.tf]: Should we remove this commented code?

File: README.md [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
 Built on top of [concourse-control-tower](https://github.com/dwp/concourse-control-tower)
 
+## Setting up on first use
+
+In order to install dependencies required by this repo, run `make bootstrap`
+
+## Deploying Concourse to Development Account
+
+```
+make boostrap-terraform-dev
```
Comment (mattburgess) [inline README.md]: typo - boostrap -> bootstrap

File: terraform/modules/concourse_web/data.tf (hunk 2 above)
Comment (dan-hill2802) [reply terraform/modules/concourse_web/data.tf]: I think they should stay as we intend to do the work shortly

File: Makefile (hunk 1 above)
Comment (dan-hill2802) [reply Makefile]: this needs to be done differently now that our admin access has changed

File: README.md (hunk 3 above)
Comment (dan-hill2802) [reply README.md]: tab predict kept this hidden from me, thanks for spotting

File: terraform/modules/vpc/variables.tf [hunk 4]
```diff
@@ -11,6 +11,20 @@ variable "tags" {
 variable "vpc" {
   description = "vpc configurables"
   type = object({
-    cidr_block = any
+    cidr_block = string
   })
 }
+variable "subnets" {
+  description = "define sizes for subnets using Terraform cidrsubnet function. Defaults suit an empty /24 VPC"
```
Comment (mark-r-m) [inline terraform/modules/vpc/variables.tf]: Is "suit" the right word here? Do you mean "The default is a /24" ??

**Pull request merged**